internal/controlplane/handlers_authz.go

```diff
@@ -148,62 +125,21 @@ func AuthorizedOnProject(ctx context.Context, projectID uuid.UUID) error {
 	return nil
 }
 
-// AuthUnaryInterceptor is a server interceptor for authentication
-func AuthUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
+// AuthorizationUnaryInterceptor is a server interceptor that sets up the user permissions
+func AuthorizationUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler) (any, error) {
 
-	opts, err := optionsForMethod(info)
-	if err != nil {
-		// Fail closed safely, rather than log and proceed.
-		return nil, status.Errorf(codes.Internal, "Error getting options for method: %v", err)
-	}
-
-	ctx = withRpcOptions(ctx, opts)
-
-	if opts.GetAnonymous() {
-		if !opts.GetNoLog() {
-			zerolog.Ctx(ctx).Info().Msgf("Bypassing authentication")
-		}
-		return handler(ctx, req)
-	}
-
-	token, err := gauth.AuthFromMD(ctx, "bearer")
-	if err != nil {
-		if statusErr, ok := status.FromError(err); ok {
-			return nil, util.FromRpcError(statusErr)
-		}
-		return nil, status.Errorf(codes.Unauthenticated, "no auth token: %v", err)
-	}
-
 	server := info.Server.(*Server)
 
-	parsedToken, err := server.vldtr.ParseAndValidate(token)
-	if err != nil {
-		return nil, status.Errorf(codes.Unauthenticated, "invalid auth token: %v", err)
+	subject := auth.GetUserSubjectFromContext(ctx)
+	if subject == nil {
+		return nil, status.Errorf(codes.Internal, "Error getting user subject")
 	}
 
 	// get user authorities from the database
 	// ignore any error because the user may not exist yet
-	authorities, _ := lookupUserPermissions(ctx, server.store, parsedToken)
-
-	if opts.GetRootAdminOnly() && !isSuperadmin(authorities) {
-		return nil, status.Errorf(codes.PermissionDenied, "user not authorized")
-	}
+	authorities, _ := lookupUserPermissions(ctx, server.store, *subject)
```

Comment:
It looks like we're ignoring the `error` return value here?  Intentional?

Reply:
It is intentional, since this interceptor is called on every request, including the one that creates a user. So if a user doesn't exist yet, then they don't have any permissions. Maybe it's better to remove the error from response?

Reply:
There are three different `return err` cases in `lookupPermissions`, and I believe it's only called from this one location.  I think it would be clearer to remove the error return if we believe we've handled all the errors within the function and don't need to do anything at the interceptor level.

Reply:
Though, I'm now seeing that we've always been ignoring that error, so I'm not going to hold this up on it.

Reply:
I removed the error from the method response